app/View/Components/Community/UserCard.php

```diff
@@ -0,0 +1,163 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\View\Components\Community;
+
+use App\Community\Enums\Rank;
+use App\Community\Enums\RankType;
+use App\Site\Enums\Permissions;
+use App\Site\Models\User as UserModel;
+use App\Support\Cache\CacheKey;
+use Illuminate\Contracts\View\View;
+use Illuminate\Support\Carbon;
+use Illuminate\Support\Facades\Cache;
+use Illuminate\View\Component;
+
+class UserCard extends Component
+{
+    private string|array $user;
+
+    public function __construct(string|array $user)
+    {
+        $this->user = $user;
+    }
+
+    public function render(): ?View
+    {
+        $username = $this->getUsername($this->user);
+
+        if (empty($username)) {
+            return null;
+        }
+
+        /* @var array $rawUserData */
+        $rawUserData = [];
+        if (is_array($this->user)) {
+            $rawUserData = $this->user;
+        } else {
+            $rawUserData = $this->getUserData($username);
+        }
+
+        if (!$rawUserData) {
+            return null;
+        }
+
+        $cardViewValues = $this->buildAllCardViewValues($username, $rawUserData);
+
+        return view('components.community.user-card', $cardViewValues);
+    }
+
+    private function getUsername(string|array $user): ?string
+    {
+        return is_string($user) ? $user : ($user['User'] ?? null);
+    }
+
+    private function getUserData(string $username): ?array
+    {
+        return Cache::store('array')->remember(
+            CacheKey::buildUserCardDataCacheKey($username),
+            Carbon::now()->addMonths(3),
+            function () use ($username): ?array {
+                $foundUser = UserModel::firstWhere('User', $username);
+
+                return $foundUser ? $foundUser->toArray() : null;
+            }
+        );
```

Comment:
I'm a bit concerned that this code is caching a `UserModel->toArray`, and the code in `userAvatar` is caching the output of `getAccountDetails` under the same cache key.

Both are using in-request caches, so there's minimal chance of the conflict actually causing issues, but you can see at least one aspect of the problem on the cards test page. If you forcibly expire the cache entry before doing the lookup in `UserCard::getUserData`, the embedded card will gain an image that's not present when it leverages the data cached by `userAvatar`.

Reply:
It is a valid concern. After studying `userAvatar()` for a moment, it appears `getAccountDetails()` is being called simply to do an existence check for the user.

This is probably overkill. I suspect something like `User::where('User', $username)->exists()` is more performant and the caching can be eliminated here entirely.

I've made this change in https://github.com/RetroAchievements/RAWeb/pull/1674/commits/da210d147e73667624ed79aea9c3cfd1d7f1f3c8.